lib/promoted/ruby/client/validator.rb

```diff
@@ -2,26 +2,151 @@ module Promoted
     module Ruby
       module Client
         class Validator
-            def self.validate_fields!(obj, obj_name, fields)
+            def validate_fields!(obj, obj_name, fields)
                 fields.each {|field|
-                    raise ValidationError.new(field[:name] + " is required on " + obj_name) if !obj.has_key?(field[:name].to_sym)
+                    if field[:required] then
+                        raise ValidationError.new(field[:name].to_s + " is required on " + obj_name) if !obj.has_key?(field[:name])
+                    end
+
+                    if field[:type] && obj.has_key?(field[:name]) then
+                        raise ValidationError.new(field[:name].to_s + " should be a " + field[:type].to_s) if !obj[field[:name]].is_a?(field[:type])
+                    end
                 }
             end
 
-            def self.validate_metrics_request!(metrics_req)
-                self.validate_fields!(
+            def validate_user_info!(ui)
+                validate_fields!(
+                    ui,
+                    "user info",
+                    [
+                        {
+                            :name => :user_id,
+                            :type => String
+                        },
+                        {
+                            :name => :log_user_id,
+                            :type => String
+                        }
+                    ]
+                )
+            end
+
+            def validate_insertion!(ins)
+                validate_fields!(
+                    ins,
+                    "insertion",
+                    [
+                        {
+                            :name => :platform_id,
+                            :type => Integer
+                        },
+                        {
+                            :name => :insertion_id,
```

Comment:
side-note - clients of the library should not set request_id, insertion_id or delivery_score.

Reply:
Ah, I think in that case a check for delivery_score needs to be added to check_that_log_ids_not_set!, which I did. Thanks.